contracts/EVMScriptFactories/AddNodeOperators.sol

```diff
@@ -0,0 +1,206 @@
+    // VARIABLES
+    // -------------
+
+    /// @notice Address of NodeOperatorsRegistry contract
+    INodeOperatorsRegistry public immutable nodeOperatorsRegistry;
+    /// @notice Address of Aragon ACL contract
+    IACL public immutable acl;
+    /// @notice Address of Lido contract
+    address public immutable lido;
+
+    // -------------
+    // ERRORS
+    // -------------
+
+    string private constant ERROR_MANAGER_ALREADY_HAS_ROLE = "MANAGER_ALREADY_HAS_ROLE";
+    string private constant ERROR_MANAGER_ADDRESSES_HAS_DUPLICATE =
+        "MANAGER_ADDRESSES_HAS_DUPLICATE";
+    string private constant ERROR_NODE_OPERATORS_COUNT_MISMATCH = "NODE_OPERATORS_COUNT_MISMATCH";
+    string private constant ERROR_LIDO_REWARD_ADDRESS = "LIDO_REWARD_ADDRESS";
+    string private constant ERROR_ZERO_REWARD_ADDRESS = "ZERO_REWARD_ADDRESS";
+    string private constant ERROR_ZERO_MANAGER_ADDRESS = "ZERO_MANAGER_ADDRESS";
+    string private constant ERROR_WRONG_NAME_LENGTH = "WRONG_NAME_LENGTH";
+    string private constant ERROR_MAX_OPERATORS_COUNT_EXCEEDED = "MAX_OPERATORS_COUNT_EXCEEDED";
+    string private constant ERROR_EMPTY_CALLDATA = "EMPTY_CALLDATA";
+
+    // -------------
+    // CONSTRUCTOR
+    // -------------
+
+    constructor(
+        address _trustedCaller,
+        address _nodeOperatorsRegistry,
+        address _acl,
+        address _lido
+    ) TrustedCaller(_trustedCaller) {
+        nodeOperatorsRegistry = INodeOperatorsRegistry(_nodeOperatorsRegistry);
+        acl = IACL(_acl);
+        lido = _lido;
+    }
+
+    // -------------
+    // EXTERNAL METHODS
+    // -------------
+
+    /// @notice Creates EVMScript to add batch of node operators
+    /// @param _creator Address who creates EVMScript
+    /// @param _evmScriptCallData Encoded (uint256,AddNodeOperatorInput[])
+    function createEVMScript(
+        address _creator,
+        bytes memory _evmScriptCallData
+    ) external view override onlyTrustedCaller(_creator) returns (bytes memory) {
+        (
+            uint256 nodeOperatorsCount,
+            AddNodeOperatorInput[] memory decodedCallData
+        ) = _decodeEVMScriptCallData(_evmScriptCallData);
+
+        address[] memory toAddresses = new address[](decodedCallData.length * 2);
+        bytes4[] memory methodIds = new bytes4[](decodedCallData.length * 2);
+        bytes[] memory encodedCalldata = new bytes[](decodedCallData.length * 2);
+
+        _validateInputData(nodeOperatorsCount, decodedCallData);
+
+        for (uint256 i = 0; i < decodedCallData.length; ++i) {
+            toAddresses[i * 2] = address(nodeOperatorsRegistry);
+            methodIds[i * 2] = ADD_NODE_OPERATOR_SELECTOR;
+            encodedCalldata[i * 2] = abi.encode(
+                decodedCallData[i].name,
+                decodedCallData[i].rewardAddress
+            );
+
+            // See https://legacy-docs.aragon.org/developers/tools/aragonos/reference-aragonos-3#parameter-interpretation for details
```

Comment:
Parameters are not intuitive, so it would be great to explain how they are encoded and what `(1 << 240)` is.